packages/react/src/SelectPanel/SelectPanel.tsx

```diff
@@ -61,6 +61,16 @@ const focusZoneSettings: Partial<FocusZoneHookSettings> = {
   disabled: true,
 }
 
+const areItemsEqual = (itemA: ItemInput, itemB: ItemInput) => {
+  // prefer checking equivality by item.id
+  if (typeof itemA.id !== 'undefined') return itemA.id === itemB.id
+  else return itemA === itemB
```

Comment:
I want to assume, this is fallback and not hopefully not used much as it doesn't solve the problem as far as I understand.

Reply:
yep, that's is the fallback. 

We would need to use it for cases without an `id` key, which works fine when the items object has a stable reference.